hal-x86_64/src/cpu.rs

```diff
@@ -1,10 +1,19 @@
 use core::mem;
+use core::fmt;
 
 #[repr(transparent)]
 pub struct Port {
     num: u16,
 }
 
+impl fmt::Debug for Port {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("Port")
+            .field("num", &format_args!("{:#02x}", self.num))
+            .finish()
+    }
+}
+
 impl Port {
     pub const fn at(address: u16) -> Self {
```

Comment:
TIOLI: it seems to me that _this_ is the actual unsafe fn, rather than `readb`/`writeb`, by calling `Port::at` you have to uphold that the address is actually a port. Once you _have_ a valid port, it is safe to read/write...

Reply:
i disagree: we don't know the semantics of the port at the given address, but we _do_ know there are 65536 i/o ports for an x86 machine. components using a specific port are then responsible for not accidentally sending bytes that might, as an example, misconfigure a device that when mixed with faulty hardware start a fire. (to be clear, this is a hypothetical, but i know of uncomfortably similar real circumstances w/ devices that don't use i/o ports)

on the other hand, just saying a port number isn't a dangerous operation.

(i think of this as, essentially, passing a pointer into a struct we later read/write through, and have about the same opinion of bare mmio whenever we get to working with that too)

Reply:
hmm...i suppose that's fair: considering a rawptr as precedent, creating a rawptr is safe, but _dereferencing_ it is unsafe, so i suppose it makes for ports to be similar. nvm.